server/logging.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import pinoLogger from 'pino';
+import pinoHttpLogger from 'pino-http';
+import { randomBytes } from 'crypto';
+
+import { loadConfig, watchConfig, getServerName } from 'serverConfig';
+import { entryExitLoggerType } from 'types';
+
+const STRIMZI_UI_REQUEST_ID_HEADER = 'x-strimzi-ui-request';
+
+const serverName = getServerName();
+
+let rootLogger = pinoLogger();
+
+/** Returns the logger used by the serverConfig functions */
+const getConfigLogger = () =>
+  rootLogger.child({
+    serverName,
+    module: 'config',
+  });
+
+/** Replaces the rootLogger with a new pinoLogger with the supplied options */
+const updateRootLoggerOptions: (
+  loggerOptions: pinoLogger.LoggerOptions,
+  isUpdate: boolean
+) => void = (loggerOptions, isUpdate) => {
+  rootLogger.info(
+    {
+      serverName,
+      module: 'logging',
+      loggerOptions,
+    },
+    `Strimzi ui logging options${isUpdate ? ' updated' : ''}`
+  );
+  try {
+    // Update the root logger with the new options
+    rootLogger = pinoLogger(loggerOptions);
+  } catch (err) {
+    rootLogger.error(
+      {
+        serverName,
+        module: 'logging',
+        loggerOptions,
+        err,
+      },
+      `Failed to configure logging options: ${err.message}`
+    );
+  }
+};
+
+/** Update the rootLogger when the config is loaded */
+loadConfig((loadedInitialConfig) => {
```

Comment:
Rather than set up/load up/watch the config in multiple places (ie here and in main.ts), is there scope to have main.ts (when it loads/changes config) call a callback provided to do the same thing by the logger?

Reply:
Yep, will see what can be done.